gems/sorbet/lib/hidden-definition-finder.rb

```diff
@@ -117,6 +117,14 @@ def write_constants
     raise "Your source can't be read by Sorbet.\nYou can try `find . -type f | xargs -L 1 -t bundle exec srb tc --no-config --error-white-list 1000` and hopefully the last file it is processing before it dies is the culprit.\nIf not, maybe the errors in this file will help: #{SOURCE_CONSTANTS_ERR}" if File.read(SOURCE_CONSTANTS).empty?
 
     puts "Printing #{TMP_RBI}'s symbol table into #{RBI_CONSTANTS}"
+
+    additional_error_blacklist = []
+    if Object.const_defined?(:RUBY_ENGINE) and RUBY_ENGINE == 'jruby'
+      # Some JRuby types have different parent classes than the ones the RBIs included
+      # with Sorbet.
+      additional_error_blacklist << '--error-black-list=5012'
```

Comment:
I'm concerned of implications of this. This will mean that Sorbet won't correctly model classes. I think the proper way here is to have some parts of standard library saved in https://github.com/sorbet/sorbet-typed and versioned so that jruby gets a different version.

Note that running Sorbet with any of `--error-black-list` isn't supported and you'll get all kinds of weird errors with them.

Reply:
Concretely this means that it could be the case that a method exists on a variable according to Sorbet, but doesn't exist at runtime in JRuby, or vice versa. I'm with Dmitry on this one.